### PR TITLE
Update the build instructions for the changes to the build tooling in or...

### PR DESCRIPTION
**Pull request by barnabycourt**
...der to independent plugin versioning

### CODE REVIEW

**Comment by beav** [comment]
This LGTM but I'd like if someone less familiar with the build stuff reviewed as well :bear:.

Setting state back to unassigned.

File: docs/dev-guide/contributing/building.rst [hunk 1]
```diff
@@ -8,54 +8,54 @@ Concepts
...
-   tags to correspond to our dev, testing, and release branch names, but at this time this is how
-   they map.
-
-For example, the 2.4 release stream in Pulp platform has three git branches, 2.4-dev, 2.4-testing,
-and 2.4-release. For RHEL 7, these map to pulp-2.4-testing-rhel7, pulp-2.4-beta-rhel7, and
-pulp-2.4-rhel7, respectively. You can see the full list of Pulp's Koji tags
+called tags. A tag is essentially a grouping of package builds.
+Pulp uses one Koji tag per Pulp X.Y release stream, per distribution, per architecture.
+For example, the 2.6 releases of pulp will build into the pulp-2.6-<distribution> tags in koji.
+You can see the full list of Pulp's Koji tags
```
Comment (rbarlow) [inline docs/dev-guide/contributing/building.rst]: I'm surprised at this change. We aren't going to use Koji beta or testing targets anymore? Was that decided when I was away? I'm not against the idea, it's just news to me.

File: docs/dev-guide/contributing/building.rst [hunk 2]
```diff
@@ -8,54 +8,54 @@ Concepts
...
-``pulp-2.4-beta-rhel7`` tag and you wish to add that exact package in the ``pulp-2.4-rhel7`` tag,
+``pulp-2.4-rhel7`` tag and you wish to add that exact package in the ``pulp-2.5-rhel7`` tag,
 you cannot build it again. Instead, you must tag that package for the new tag. You will see later
 on in this document that Pulp has a tool to help you do this.
 
+Pulp release and testing builds are collections of components that are versioned independently.
+For example, the core Pulp server may be at version 2.6 while pulp_docker may be at version 1.0.
+This assembly is accomplished using release definitions specified in the
+``pulp_packaging/ci/config/releases/<build-name>.yaml`` files. Each file specifies the detail of a
+build that the Pulp build scripts can later can assemble. The components within that
```
Comment (rbarlow) [inline docs/dev-guide/contributing/building.rst]: s/can assemble/assemble/

File: docs/dev-guide/contributing/building.rst [hunk 3]
```diff
@@ -8,54 +8,54 @@ Concepts
...
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-Pulp has some wrapper scripts in the ``pulp/rel-eng`` directory to assist with
+Test or release builds (with the exclusion of the signing step) may be performed using
+Jenkins.  There are automated jobs that will run nightly which build repositories that can be used
+for validation.  When those jobs are initiated manually there is a parameter to enable the
+release build process in koji.  If a release build is performed with Jenkins you will still need
+to sign the rpms and push them to the final, non automation, location on fedorapeople.
+
+Pulp has some helper scripts in the ``pulp_packaging/ci`` directory to assist with
```
Comment (rbarlow) [inline docs/dev-guide/contributing/building.rst]: Maybe we could make the pulp_packaging reference a hyperlink to that repository.

File: docs/dev-guide/contributing/building.rst [hunk 4]
```diff
@@ -8,54 +8,54 @@ Concepts
...
+
 Tools used when building
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-Pulp has some wrapper scripts in the ``pulp/rel-eng`` directory to assist with
+Test or release builds (with the exclusion of the signing step) may be performed using
+Jenkins.  There are automated jobs that will run nightly which build repositories that can be used
+for validation.  When those jobs are initiated manually there is a parameter to enable the
+release build process in koji.  If a release build is performed with Jenkins you will still need
+to sign the rpms and push them to the final, non automation, location on fedorapeople.
```
Comment (rbarlow) [inline docs/dev-guide/contributing/building.rst]: The ", non automation," bit reads a little strangely to me. Maybe we could just say:

"and manually push them to the final location on fedorapeople."

File: docs/dev-guide/contributing/building.rst [hunk 5]
```diff
@@ -176,61 +175,34 @@ new tag to github.
...
-
-.. note::
+Test Building Pulp and the plugins
+----------------------------------
 
-   The above command will finish quickly, but it will tell you that you need to manually monitor
-   Koji and wait for the repository building tasks to complete. You can view
-   `active Koji tasks <http://koji.katello.org/koji/tasks>`_. Do not submit any new Koji tasks until
-   these complete.
+Are you ready to actually build something?  If so, you should `cd` to the ``pulp_packaging/ci``
```
Comment (rbarlow) [inline docs/dev-guide/contributing/building.rst]: s/actually//

File: docs/dev-guide/contributing/building.rst [hunk 6]
```diff
@@ -260,59 +226,50 @@ stream you are building.
...
-
-Pay attention to the instructions from tito, as you will need to push your changes to the upstream Pulp
-repository, as well as the tags that tito generated.
+   $ cd /tmp/pulp_build/pulp_packaging/ci/
+   $ ./build-all.py 2.6-testing --show-versions
+   .. Versions displayed ..
+   $ ./update-version.py --update-type stage /tmp/pulp_build/pulp_packaging
 
+At this point you can inspect the files to ensure the versions are as you expect. The changes
+will still need to be committed & pushed to github before they can be used to build. You can
```
Comment (rbarlow) [inline docs/dev-guide/contributing/building.rst]: s/&/and/

File: docs/dev-guide/contributing/building.rst [hunk 7]
```diff
@@ -260,59 +226,50 @@ stream you are building.
...
+rerun the ``build-all.py`` script to check the versions again after your changes have been
+pushed to github.
```
Comment (rbarlow) [inline docs/dev-guide/contributing/building.rst]: We should probably use "GitHub" instead of "github" throughout this document.

**Comment by rbarlow** [comment]
Aside from the bit that is news to me, LGTM.

File: docs/dev-guide/contributing/building.rst (hunk 1 above)
Comment (barnabycourt) [inline docs/dev-guide/contributing/building.rst]: Not so much formally decided so much as that with the new assembly method they just were no longer providing value.

**Pull request merged**